extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/hll/sql/HllSketchSqlAggregatorTest.java

```diff
@@ -1318,6 +1320,20 @@ public void testFloatAndDoubleAreConsideredTheSame()
     );
   }
 
+  @Test
+  public void testDsHllOnTopOfNested()
+  {
+    // this query was not planable: https://github.com/apache/druid/issues/15353
+    testBuilder()
+        .queryContext(ImmutableMap.of(QueryContexts.ENABLE_DEBUG, true))
+        .sql(
+            "SELECT d1,dim2,APPROX_COUNT_DISTINCT_DS_HLL(dim2, 18) as val"
+                + " FROM (select d1,dim1,dim2 from druid.foo group by d1,dim1,dim2 order by dim1 limit 3) t "
+                + " group by 1,2"
+        )
+        .run();
```

Comment:
We should add the expected results of this query as well to make sure the results do not change later down the line

Reply:
yeah - I've added it! :)
thank you for taking a look!